### PR TITLE
add tests for reports where OpenShift versions have been edited

### PR DESCRIPTION
**Pull request by mmulholla**
Added tests which detect when versions in a  report have been. Also improved the messages in the code. Tests are quick as they complete without installing and testing the chart. Four scenarios:

1. SupportedOpenShiftVersions is not a valid semantic version.
2. Tested OpenShift version pre-dates helm support.
3. Tested OpenShift version is not in the range of the kube version.
4. kubeVersion and SupportedOpenShiftVersions are not equivalent.

https://issues.redhat.com/browse/HELM-279

### CODE REVIEW

**Review by tisutisu — COMMENTED** [review]
Overall LGTM, except the community path test scenario is missing.

File: tests/functional/features/report_only_edited.feature [hunk 1]
```diff
@@ -0,0 +1,23 @@
+Feature: Edited report only submission
+  Partners, redhat and community users attempt to publish their chart by submitting
+  report that was edited afyter it was generated by chart-verifier.
+
+  Examples:
+  | vendor_type  | vendor    | report_path               |
+  | partners     | hashicorp | tests/data/report.yaml    |
+  | redhat       | redhat    | tests/data/report.yaml    |
+
```
[the remark this answers is not in the document]
Comment (mmulholla) [reply tests/functional/features/report_only_edited.feature:10]: I added community test.

File: tests/functional/features/report_only_edited.feature [hunk 2]
```diff
@@ -0,0 +1,24 @@
+Feature: Edited report only submission
+  Partners, redhat and community users attempt to publish their chart by submitting
+  report that was edited afyter it was generated by chart-verifier.
```
Comment (zonggen) [inline tests/functional/features/report_only_edited.feature:3]:
```suggestion
  report that was edited after it was generated by chart-verifier.
```

File: tests/functional/features/report_only_edited.feature [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    Given the vendor <vendor> has a valid identity as <vendor_type>
+    And a <report_path> is provided
+    And the report includes <tested> and <supported> OpenshiftVersion values and chart <kubeversion> value
+    When the user sends a pull request with the report
+    Then the pull request is not merged
+    And user gets the <message> in the pull request comment
+
+    Examples:
+      | tested | supported | kubeversion | message                                   |
+      | 4.9    | 4.6-4.9   | >=1.20.0    | is not a valid semantic version           |
```
Comment (zonggen) [inline tests/functional/features/report_only_edited.feature:21]: I thought we handle a semver range like `4.6-4.9`, which part of this case is an invalid semver?

**Comment by zonggen** [comment]
LGTM other than typo and a question.

File: tests/functional/features/report_only_edited.feature (hunk 3 above)
Comment (mmulholla) [reply tests/functional/features/report_only_edited.feature:21]: ```4.6 - 4.9```  is a valid semver range - it requires the spaces.

File: tests/functional/features/report_only_edited.feature (hunk 2 above)
Comment (mmulholla) [reply tests/functional/features/report_only_edited.feature]: Fixed, thanks

File: tests/functional/features/report_only_edited.feature (hunk 1 above)
[the remark this answers is not in the document]
Comment (tisutisu) [reply tests/functional/features/report_only_edited.feature:10]: Thanks for the update. LGTM now.

**Review by zonggen — APPROVED** [review]
LGTM

**Pull request merged**